src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -562,17 +574,24 @@ export async function refundRequestIsValid(
     return { valid: false, reason: "Unable to find matching deposit" };
   }
 
-  // Verify that the refundToken maps to a known HubPool token.
-  // Note: the refundToken must be valid at the time of the Fill *and* the RefundRequest.
-  // @todo: Resolve to the HubPool block number at the time of the RefundRequest ?
-  const hubPoolBlockNumber = hubPoolClient.latestBlockNumber ?? hubPoolClient.deploymentBlock - 1;
+  // Verify that the refundToken maps to a known HubPool token and is the correct
+  // token for the chain where the refund was sent from.
+  // Note: the refundToken must be valid at the time the deposit was sent.
   try {
-    hubPoolClient.getL1TokenCounterpartAtBlock(repaymentChainId, refundToken, hubPoolBlockNumber);
+    const l1TokenForFill = hubPoolClient.getL1TokenCounterpartAtBlock(
+      fill.destinationChainId,
+      fill.destinationToken,
+      deposit.quoteBlockNumber
```

Comment:
Nice, the quote block number is a good fit for this.

Reply:
It's not only a good fit it's the only correct block to insert here. The previous block passed was incorrect